components/external_storage/export/src/grpc_service.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::io::{self, ErrorKind};
+use std::sync::Arc;
+
+use crate::request::{restore_receiver, write_receiver};
+use anyhow::Context;
+use external_storage::request::anyhow_to_io_log_error;
+use grpcio::{self};
+use kvproto::backup as proto;
+use slog_global::{error, info};
+use tokio::runtime::{Builder, Runtime};
+
+#[derive(Debug)]
+pub struct SocketService {
+    server: grpcio::Server,
+    listener: std::os::unix::net::UnixListener,
+}
+
+pub fn new_service() -> io::Result<SocketService> {
+    (|| -> anyhow::Result<SocketService> {
+        let env = Arc::new(grpcio::EnvBuilder::new().build());
+        let storage_service = Service::new().context("new storage service")?;
+        let builder = grpcio::ServerBuilder::new(env)
+            .register_service(proto::create_external_storage(storage_service));
+        let grpc_socket_path = "/tmp/grpc-external-storage.sock";
+        let socket_addr = format!("unix:{}", grpc_socket_path);
+        let socket_path = std::path::PathBuf::from(grpc_socket_path);
+        // Keep the listener in scope: otherwise the socket is destroyed
+        let listener = bind_socket(&socket_path).context("GRPC new service create socket")?;
```

Comment:
Is it in-process RPC?

Reply:
No. For in process one would use the C dynamic lib.

Reply:
The idea behind RPC is that the server is ran as a separate process: thus the need for the socket.